exampleConnectionService/src/main/java/com/twilio/voice/examples/connectionservice/VoiceActivity.java

```diff
@@ -0,0 +1,337 @@
+
+import java.util.HashMap;
+
+public class VoiceActivity extends AppCompatActivity {
+    private static final String TAG = "VoiceActivity";
+
+    /*
+     * You must provide the URL to the publicly accessible Twilio access token server route
+     *
+     * For example: https://myurl.io/accessToken
+     */
+    private static final String TWILIO_ACCESS_TOKEN_SERVER_URL = "TWILIO_ACCESS_TOKEN_SERVER_URL";
+
+    public static final String OUTGOING_CALL_ADDRESS = "OUTGOING_CALL_ADDRESS";
+
+    public static final String ACTION_OUTGOING_CALL = "ACTION_OUTGOING_CALL";
+    public static final String ACTION_DISCONNECT_CALL = "ACTION_DISCONNECT_CALL";
+    public static final String ACTION_DTMF_SEND = "ACTION_DTMF_SEND";
+    public static final String DTMF = "DTMF";
+    public static final String CALLEE = "to";
+
+    private static final int MIC_PERMISSION_REQUEST_CODE = 1;
+    private static final int CALL_PHONE_CODE = 2;
+    private static final int SNACKBAR_DURATION = 4000;
+    private CoordinatorLayout coordinatorLayout;
+
+    private String accessToken;
+    private boolean isReceiverRegistered = false;
+    private VoiceBroadcastReceiver voiceBroadcastReceiver;
+
+    int PERMISSION_ALL = 1;
+    String[] PERMISSIONS = {Manifest.permission.RECORD_AUDIO, Manifest.permission.CALL_PHONE};
+
+    private TelecomManager telecomManager;
+    private PhoneAccountHandle handle;
+    private PhoneAccount phoneAccount;
+
+    HashMap<String, String> twiMLParams = new HashMap<>();
+    private AlertDialog alertDialog;
+    Call activeCall;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_voice);
+        coordinatorLayout = findViewById(R.id.coordinator_layout);
+        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
+        setSupportActionBar(toolbar);
+
+        FloatingActionButton fab = (FloatingActionButton) findViewById(R.id.fab);
+        fab.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                alertDialog = createCallDialog(callClickListener(), cancelCallClickListener(), VoiceActivity.this);
+                alertDialog.show();
+            }
+        });
+
+        Voice.setLogLevel(LogLevel.ALL);
+        Voice.setModuleLogLevel(LogModule.PJSIP, LogLevel.ALL);
+
+        /*
+         * Ensure the microphone and CALL_PHONE permissions are enabled
+         */
+        if (!hasPermissions(this, PERMISSIONS)) {
+            ActivityCompat.requestPermissions(this, PERMISSIONS, PERMISSION_ALL);
+        } else {
+            retrieveAccessToken();
+        }
+    }
+
+    private void init() {
+        setupPhoneAccount();
+        voiceBroadcastReceiver = new VoiceBroadcastReceiver();
+        registerReceiver();
+    }
+
+    /*
+    * Setup a phone Account
+    */
+    void setupPhoneAccount() {
```

Comment:
Move `init()` to before requesting permissions or retrieving an access token in `onCreate(...)` since it's not dependent on either.

Reply:
Rather than have a function just inline those 3 lines in the `onCreate(...)`